### ISSUE TITLE
Private IP protection Bypass in private-ip package 

### ISSUE THREAD

**Issue by x3rz** [frenchbread/private-ip#14]
Hello @frenchbread  i am x3rz a researcher on huntr.dev as you wanted to discuss about this issue here 

Yes it gives 'false' because they are resolved to private IP as the main objective of the package is telling that the IP is private or not. As you said about 127.000.000.1 so basically this resolves to localhost which is private IP and also for the other payloads especially like decimal encoded of localhost value i.e 2130706433 gives false but if you visit this from your browser you will see that this also resolves to localhost but for the hex-encoded payload for localhost package works just fine 0x7f000001 gives "true" for this one. And Yes you can resolve it to undefined and stop the further execution as similar to 'false' but as these all payloads resolve to localhost which could cause ssrf For more refrences you can check This one which contains all the payloads that resolves to private IP addresses https://github.com/swisskyrepo/PayloadsAllTheThings/tree/master/Server%20Side%20Request%20Forgery#file This one is just for reference that a blocklist could be implemented which could handel this issue pretty easily https://github.com/y-mehta/ssrf-req-filter/blob/master/test/blockUrls.txt This one shows the possible bypasses and the logic behind them https://vickieli.medium.com/bypassing-ssrf-protection-e111ae70727b

**Comment by x3rz** [comment]
I have added one more case in the cases of runkit for the hex encoded payload so that it can be easily identified which implies that 127.0.0.1 = Dec(2130706433) = Hex(0x7f000001)

**Comment by frenchbread** [comment]
Hi, @x3rz. Thanks for opening an issue here.

**Comment by frenchbread** [comment]
Hey, @x3rz, I've created a PR that handles payload patterns you've provided.

Would you be able to test it & confirm or suggest further improvements needed?

Thanks!

**Comment by x3rz** [comment]
Hey, @frenchbread, Sorry for the late response,
Sure I will but for ease can you please specify steps I need to follow to set up the module on my local system as I am facing some issues with lib.

**Comment by x3rz** [comment]
Sorry for the late reply, I have tested all the issues and all tests are passed, as this only tells IP is private or not.
If you gonna extended this package please do consider all the IPs listed in 
https://gist.github.com/x3rz/8f56d8832dd6965b02867900f2ade9ca
Thank you @frenchbread

**Comment by frenchbread** [comment]
@x3rz thanks for reply,

I addition to `is_private` => `true` or `false`, module also returns `undefined` for ips that neither is ip nor follow ipv4/ipv6 schema.

I've checked out the link and I assume this is not really in scope of this module (at least for now). This PR already adds support for handling `dec` and `hex` payloads for ipv4.

I'll leave this PR open.

**Comment by x3rz** [comment]
yes exactly what i said and the issue I reported should get patched as per my pov and you can keep this opened 

Regards x3rz

**Comment by x3rz** [comment]
Hello @frenchbread
I recently applied for CVE for previous package in which i found those bypasses and need your acknowledgement. 

Regards,
x3rz

**Comment by frenchbread** [comment]
Hi @x3rz,

Thanks for letting me know!
